tests/core/pyspec/eth2spec/test/altair/sync_protocol/test_light_client_sync.py

```diff
@@ -0,0 +1,424 @@
+    test.genesis_validators_root = state.genesis_validators_root
+
+    next_slots(spec, state, spec.SLOTS_PER_EPOCH * 2 - 1)
+    trusted_block = state_transition_with_full_block(spec, state, True, True)
+    trusted_block_root = trusted_block.message.hash_tree_root()
+    bootstrap = spec.create_light_client_bootstrap(state)
+    yield "trusted_block_root", "meta", "0x" + trusted_block_root.hex()
+    yield "bootstrap", bootstrap
+    test.store = spec.initialize_light_client_store(trusted_block_root, bootstrap)
+
+    return test
+
+
+def finalize_test(test):
+    yield "steps", test.steps
+    yield "expected_finalized_header", test.store.finalized_header
+    yield "expected_optimistic_header", test.store.optimistic_header
+
+
+def get_update_file_name(spec, update):
+    if spec.is_sync_committee_update(update):
+        suffix1 = "s"
+    else:
+        suffix1 = "x"
+    if spec.is_finality_update(update):
+        suffix2 = "f"
+    else:
+        suffix2 = "x"
+    return f"update_{encode_hex(update.attested_header.hash_tree_root())}_{suffix1}{suffix2}"
+
+
+def emit_slot(test, spec, state):
+    current_slot = state.slot
+    yield from []
+    test.steps.append({
+        "process_slot": {
+            "current_slot": int(current_slot),
+        }
+    })
+    spec.process_slot_for_light_client_store(test.store, current_slot)
+
+
+def emit_update(test, spec, state, block, attested_state, finalized_block, with_next_sync_committee=True):
+    update = spec.create_light_client_update(state, block, attested_state, finalized_block)
+    if not with_next_sync_committee:
+        update.next_sync_committee = spec.SyncCommittee()
+        update.next_sync_committee_branch = \
+            [spec.Bytes32() for _ in range(spec.floorlog2(spec.NEXT_SYNC_COMMITTEE_INDEX))]
+    current_slot = state.slot
+    yield get_update_file_name(spec, update), update
+    test.steps.append({
+        "process_update": {
+            "update": get_update_file_name(spec, update),
+            "current_slot": int(current_slot),
+        }
+    })
+    spec.process_light_client_update(test.store, update, current_slot, test.genesis_validators_root)
+    return update
+
+
+def compute_start_slot_at_sync_committee_period(spec, sync_committee_period):
+    return spec.compute_start_slot_at_epoch(sync_committee_period * spec.EPOCHS_PER_SYNC_COMMITTEE_PERIOD)
+
+
+def compute_start_slot_at_next_sync_committee_period(spec, state):
+    sync_committee_period = spec.compute_sync_committee_period(spec.compute_epoch_at_slot(state.slot))
+    return compute_start_slot_at_sync_committee_period(spec, sync_committee_period + 1)
+
+
+@with_altair_and_later
+@spec_state_test_with_matching_config
+@with_presets([MINIMAL], reason="too slow")
+def test_light_client_sync(spec, state):
+    if state.fork.current_version == spec.config.GENESIS_FORK_VERSION:
+        return  # Test requires Altair or later
```

Comment:
We don't need this check because we have `@with_altair_and_later` to filter pre-Altair forks out. Same for rest tests.

```suggestion
```

Reply:
Yeah, that's true for `make citest fork=altair`, but `make test` still runs this for `GENESIS_FORK_VERSION`. For now, I have extended the comment to highlight this oddity: 73279f8382230bc3eb5a48ffd28ac0c27f57b359

Reply:
> `make test` still runs this for `GENESIS_FORK_VERSION`

I did some debug, the error happened in Capella at [`assert compute_epoch_at_slot(state.slot) >= ALTAIR_FORK_EPOCH`](https://github.com/ethereum/consensus-specs/pull/2938/files#diff-4a04d8e702797700bb33e390b2ab32b29f8797819602cd5cb8cead73d1045c3aR43) in `create_light_client_bootstrap`.

It turns out the bug is in the Capella genesis setting! Fixed it in #2940. Once #2940 is merged, we should be able to remove these lines. Sorry for the inconvenience. 😅

Reply:
261c5317e1929aaf64d0029ddb306308b0020ea2